src/Raven.Server/Documents/Queries/Dynamic/DynamicQueryRunner.cs

```diff
@@ -185,6 +185,13 @@ private async Task<(Index Index, string Collection)> MatchIndex(IndexQueryServer
         {
             var collection = query.Metadata.CollectionName;
 
+            if (query.Metadata.DynamicIndexName != null)
```

Comment:
1. I'm assuming that as long as index exists, it can be used to get the query results.
2. You are right, we cannot share metadata between the databases.